examples/step-87/step-87.cc

```diff
@@ -0,0 +1,1196 @@
+              AffineConstraints<double>().distribute_local_to_global(
+                local_values, local_dof_indices, force_vector);
+            }
+        };
+
+        // The callback function is passed together with the vector holding the
+        // surface-tension force contribution at each quadrature point of the
+        // immersed mesh to
+        // Utilities::MPI::RemotePointEvaluation::process_and_evaluate. The only
+        // missing step is to compress the distributed force vector.
+        rpe.process_and_evaluate<value_type>(integration_values,
+                                             integration_function);
+        force_vector.compress(VectorOperation::add);
+
+        // After every tenth step or at the beginning/end of the time loop, we
+        // output the force vector and the velocity of the background mesh to
+        // a VTU file. In addition, we also export the geometry of the
+        // (deformed) immersed surface mesh to a separate VTU file.
+        if (it % 10 == 0 || it == n_time_steps)
+          {
+            std::vector<
+              DataComponentInterpretation::DataComponentInterpretation>
+              vector_component_interpretation(
+                dim, DataComponentInterpretation::component_is_part_of_vector);
+            pcout << "  - write data (background mesh)" << std::endl;
+            DataOut<dim>          data_out_background;
+            DataOutBase::VtkFlags flags_backround;
+            flags_backround.write_higher_order_cells = true;
+            data_out_background.set_flags(flags_backround);
+            data_out_background.add_data_vector(
+              dof_handler_background,
+              force_vector,
+              std::vector<std::string>(dim, "force"),
+              vector_component_interpretation);
+            data_out_background.add_data_vector(
+              dof_handler_background,
+              velocity,
+              std::vector<std::string>(dim, "velocity"),
+              vector_component_interpretation);
+            data_out_background.build_patches(mapping_background, 3);
+            data_out_background.write_vtu_in_parallel("example_3_background_" +
+                                                        std::to_string(it) +
+                                                        ".vtu",
+                                                      MPI_COMM_WORLD);
+
+            pcout << "  - write mesh (immersed mesh)" << std::endl;
+            DataOut<dim - 1, dim> data_out_immersed;
+            data_out_immersed.attach_triangulation(tria_immersed);
+            data_out_immersed.build_patches(mapping_immersed, 3);
+            data_out_immersed.write_vtu_in_parallel("example_3_immersed_" +
+                                                      std::to_string(it) +
+                                                      ".vtu",
+                                                    MPI_COMM_WORLD);
```

Comment:
This could be moved to a function (maybe it would increase readability), but if you find you need to pass too many variables as arghument to that function, then maybe it is not useful

Reply:
I agree but I am afraid that given that we don't have classes with according member variables an extra function would need too many input parameters as to improve the readability.